indico/modules/events/papers/controllers/display.py

```diff
@@ -50,6 +54,41 @@ def _process(self):
         return jsonify_form(form, form_header_kwargs={'action': request.relative_url})
 
 
+class RHDownloadPaperFile(RHPapersBase):
+    """Download a paper file"""
+
+    normalize_url_spec = {
+        'locators': {
+            lambda self: self.paper_file
+        }
+    }
+
+    def _checkParams(self, params):
+        RHPapersBase._checkParams(self, params)
+        self.paper_file = PaperFile.get_one(request.view_args['file_id'])
+
+    def _process(self):
+        return self.paper_file.send()
+
+
 class RHPaperTimeline(RHPapersBase):
     def _process(self):
         return NotImplementedError
+
+
+class RHDisplayReviewingArea(RHPapersBase):
+    def _checkProtection(self):
+        if not session.user:
+            raise Forbidden
+            RHPapersBase._checkProtection(self)
```

Comment:
Also, this should check whether the user is a reviewer/judge in the event. Being logged in isn't enough to use that page (even though it'll be empty if one is not a judge/reviewer)